src/contrib/mlir/lowerer.cpp

```diff
@@ -858,6 +876,51 @@ namespace
                     NGRAPH_CHECK(false, "Unsupported type for Negative");
                 }
             }
+            else if (isa<NGNotOp>(op))
+            {
+                iRes(ivs) = !val;
+            }
+            else if (isa<NGAbsOp>(op))
+            {
+                if (auto floatTy = elemTy.dyn_cast<FloatType>())
+                {
+                    ValueHandle zero = intrinsics::constant_float(llvm::APFloat(0.0f), floatTy);
+                    iRes(ivs) = edsc::intrinsics::select((val < zero), zero - val, val);
```

Comment:
Abs for floating point would also need to call the math function counterpart to do the right thing for special values (NaN, 0.0, etc.). I guess it's fine for now but please add a note about it so that it's addressed with the other math ops.